### PR TITLE
save: fix saving not updating sg format

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes

#### Descripton

#442

### CODE REVIEW

**Comment by walkawayy** [comment]
Could you publish a build here so @Richard-L and @rapora9 can test? I'm unable to currently.

**Comment by walkawayy** [comment]
Ok I found the old publish release command. But the build fails:
```
| FAILED: Tomb1Main.exe.p/src_game_savegame.c.obj 
| /usr/bin/i686-w64-mingw32-gcc -ITomb1Main.exe.p -I. -I.. -I../src -I/ext//include -I/ext/include/SDL2 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -O3 -Wno-unused -D_GNU_SOURCE '-DVERSION="T1M 2.6.3-1-ga1e5ad3"' -Dmain=SDL_main -MD -MQ Tomb1Main.exe.p/src_game_savegame.c.obj -MF Tomb1Main.exe.p/src_game_savegame.c.obj.d -o Tomb1Main.exe.p/src_game_savegame.c.obj -c ../src/game/savegame.c
| ../src/game/savegame.c: In function ‘Savegame_Save’:
| ../src/game/savegame.c:395:26: error: ‘GAME_INFO’ {aka ‘struct GAME_INFO’} has no member named ‘format’
|   395 |                 game_info->format = strategy->format;
|       |                          ^~
```

File: src/game/savegame.c [hunk 1]
```diff
@@ -392,6 +392,8 @@ bool Savegame_Save(int32_t slot_num, GAME_INFO *game_info)
             MYFILE *fp = File_Open(full_path, FILE_OPEN_WRITE);
             if (fp) {
                 strategy->save_to_file(fp, game_info);
+                game_info->format = strategy->format;
+                game_info->current_save_slot = slot_num;
```
Comment (walkawayy) [inline src/game/savegame.c:396]: `GAME_INFO` doesn't have a `format`.

**Review by walkawayy — CHANGES_REQUESTED** [review]

**Comment by walkawayy** [comment]
[Tomb1Main-2.6.3-1-g3a6e85d_test_death.zip](https://github.com/rr-/Tomb1Main/files/8100725/Tomb1Main-2.6.3-1-g3a6e85d_test_death.zip)

**Comment by Rapora9** [comment]
This time there is no error message at all, the game just crashes.
What I tested: load legacy save -> save into an empty slot -> die.

**Comment by rr-** [comment]
[Tomb1Main-2.6.3-1-gb4df71d.zip](https://github.com/rr-/Tomb1Main/files/8100925/Tomb1Main-2.6.3-1-gb4df71d.zip)

**Review by walkawayy — APPROVED** [review]

**Comment by walkawayy** [comment]
LGTM

**Pull request merged**